classes/submission/Repository.php

```diff
@@ -837,6 +839,51 @@ public function getDashboardViews(Context $context, User $user, array $selectedR
         return $filteredViews;
     }
 
+    /**
+     * Return a collection of all existing Version Data 
+     * for the submission's publications
+     *
+     * @return Collection<VersionData>
+     */
+    public function getAllVersionDataByPublications(Submission $submission): Collection
+    {
+        return collect($submission->getData('publications')->map(function ($publication) {
+            return $publication->getCurrentVersionData();
+        })->filter()); // Remove any null entries from the collection
+    }
+
+    public function getNextAvailableVersionData(Submission $submission, VersionStage $versioningStage, bool $isMinorChange = true): VersionData
+    {
+        $nextVersionStage = VersionData::createDefaultForStage($versioningStage);
+
+        $submissionVersionStages = $this->getAllVersionDataByPublications($submission);
+
+        // Filter to find the version stages that match the current versioningStage but with potentially updated versionMajor or versionMinor
+        $matchingVersionStages = $submissionVersionStages->filter(function ($existingVersionStage) use ($versioningStage) {
```

Comment:
Should be `versionStage` rather than `versioningStage` here and below, I think.

Reply:
Yes it should - changed here and elsewhere